doc/ck_pr_fence_lock

```diff
@@ -36,79 +36,36 @@ Concurrency Kit (libck, \-lck)
 .Ft void
 .Fn ck_pr_fence_lock void
 .Sh DESCRIPTION
-This function enforces the ordering of any memory operations preceding and related to
-the acquisition of a lock relative to the invocation of the function.
-Any realted operations that were performed prior to the invocation of
-.Fn ck_pr_fence_lock
-will complete before any subsequent related operations inside the critical section.
-Because of this, 
-.Fn ck_pr_fence_lock
-will require less overhead than a traditional full fence such as mfence
-as the compiler is still free to order around the specified operations.
-This function always serves as an implicit compiler barrier.
-On architectures with CK_MD_TSO or CK_MD_PSO specified (total store ordering
-and partial store ordering respectively), this operation only serves
-as a compiler barrier and no fence instructions will be emitted. To
-force the unconditional emission of a fence, use
+This function serves as an optimization surrounding atomic lock operations.
```

Comment:
Let's drop this sentence as the purpose is to enforce ordering, it is more optimal than other options, but that is still it's purpose.